### PR TITLE
Apply workflow guidelines to Amor

### LINKED ISSUE #48: Workflow guidelines in amor reduction
Making a list of things to do before I forget:

1. Use an `AmorWorkflow` with default parameters and providers, instead of talking about providers in the notebook.

2. Naming (in accordance with SANS and diffraction workflows):
  - `Run` -> `RunType`
  - `Sample` -> `SampleRun`
  - What to do about `FilePath` vs dedicated functions to provide file names
  - `RawDetector` -> `LoadedNeXusDetector` ?
  - `RawEvents` -> `RawDetectorData`
  - `ChopperCorrectedTofEvents` -> `ReducibleDetectorData` ?

### PR DESCRIPTION
**Pull request by nvaytet**
Fixes #48

### CODE REVIEW

File: src/ess/reflectometry/orso.py [hunk 1]
```diff
@@ -32,102 +32,102 @@
...
+OrsoReduction = NewType("OrsoReduction", reduction.Reduction)
 """ORSO data reduction metadata."""
 
-OrsoSample = NewType('OrsoSample', data_source.Sample)
+OrsoSample = NewType("OrsoSample", data_source.Sample)
 """ORSO sample."""
 
 
-def parse_orso_experiment(filepath: FilePath[Sample]) -> OrsoExperiment:
+def parse_orso_experiment(Filename: Filename[SampleRun]) -> OrsoExperiment:
```
Comment (jokasimr) [inline src/ess/reflectometry/orso.py:63]: (minor) should be lowercase right?

File: src/ess/reflectometry/orso.py [hunk 2]
```diff
@@ -32,102 +32,102 @@
...
             facility=facility,
             start_date=parse_datetime(start_time),
-            probe='neutron',
+            probe="neutron",
         )
     )
 
 
-def parse_orso_owner(filepath: FilePath[Sample]) -> OrsoOwner:
+def parse_orso_owner(Filename: Filename[SampleRun]) -> OrsoOwner:
```
Comment (jokasimr) [inline src/ess/reflectometry/orso.py:83]: is the case thing something I missed that is part of the guidelines?

**Review by jokasimr — APPROVED** [review]
Looks good to me!

File: src/ess/reflectometry/orso.py (hunk 1 above)
Comment (nvaytet) [reply src/ess/reflectometry/orso.py:63]: I think it's a bad find/replace in the editor...

**Pull request merged**